src/plugins/interactive_setup/server/elasticsearch_service.test.ts

```diff
@@ -0,0 +1,497 @@
+        await nextTick();
+
+        expect(mockConnectionStatusClient.asInternalUser.ping).toHaveBeenCalledTimes(1);
+        expect(mockHandler).toHaveBeenCalledTimes(1);
+        expect(mockHandler).toHaveBeenCalledWith(ElasticsearchConnectionStatus.NotConfigured);
+
+        mockSubscription.unsubscribe();
+        mockHandler.mockClear();
+        mockConnectionStatusClient.asInternalUser.ping.mockClear();
+
+        jest.advanceTimersByTime(5000);
+        await nextTick();
+
+        expect(mockConnectionStatusClient.asInternalUser.ping).not.toHaveBeenCalled();
+        expect(mockHandler).not.toHaveBeenCalled();
+
+        jest.advanceTimersByTime(5000);
+        await nextTick();
+
+        expect(mockConnectionStatusClient.asInternalUser.ping).not.toHaveBeenCalled();
+        expect(mockHandler).not.toHaveBeenCalled();
+      });
+
+      it('treats non-connection errors the same as successful response', async () => {
+        mockConnectionStatusClient.asInternalUser.ping.mockRejectedValue(
+          new errors.ResponseError(
+            interactiveSetupMock.createApiResponse({ statusCode: 401, body: {} })
+          )
+        );
+
+        const mockHandler = jest.fn();
+        setupContract.connectionStatus$.subscribe(mockHandler);
+
+        jest.advanceTimersByTime(0);
+        await nextTick();
+
+        expect(mockConnectionStatusClient.asInternalUser.ping).toHaveBeenCalledTimes(1);
+        expect(mockHandler).toHaveBeenCalledTimes(1);
+        expect(mockHandler).toHaveBeenCalledWith(ElasticsearchConnectionStatus.Configured);
+
+        mockHandler.mockClear();
+        mockConnectionStatusClient.asInternalUser.ping.mockClear();
+
+        jest.advanceTimersByTime(5000);
+        await nextTick();
+
+        expect(mockConnectionStatusClient.asInternalUser.ping).not.toHaveBeenCalled();
+        expect(mockHandler).not.toHaveBeenCalled();
+      });
+
+      it('treats product check error the same as successful response', async () => {
+        mockConnectionStatusClient.asInternalUser.ping.mockRejectedValue(
+          new errors.ProductNotSupportedError(interactiveSetupMock.createApiResponse({ body: {} }))
+        );
+
+        const mockHandler = jest.fn();
+        setupContract.connectionStatus$.subscribe(mockHandler);
+
+        jest.advanceTimersByTime(0);
+        await nextTick();
+
+        expect(mockConnectionStatusClient.asInternalUser.ping).toHaveBeenCalledTimes(1);
+        expect(mockHandler).toHaveBeenCalledTimes(1);
+        expect(mockHandler).toHaveBeenCalledWith(ElasticsearchConnectionStatus.Configured);
+
+        mockHandler.mockClear();
+        mockConnectionStatusClient.asInternalUser.ping.mockClear();
+
+        jest.advanceTimersByTime(5000);
+        await nextTick();
+
+        expect(mockConnectionStatusClient.asInternalUser.ping).not.toHaveBeenCalled();
+        expect(mockHandler).not.toHaveBeenCalled();
+      });
+    });
+
+    describe('#enroll()', () => {
```

Comment:
__note:__ quite a bit of "testing of internal implementation" here, but I feel like it's super important to not break anything in this area.